server/data/userVersionMiddleware.ts

```diff
@@ -0,0 +1,18 @@
+/* istanbul ignore file */
+
+import { Response, SuperAgentRequest } from 'superagent'
+import inMemoryStore from '../inMemoryStore'
+
+function userVersionMiddleware(agent: SuperAgentRequest) {
+  agent.on('request', ({ req }) => {
+    req.on('response', (res: Response, err: Error) => {
+      res.on('end', () => {
+        inMemoryStore.userVersion = res.headers['x-cas-user-version']
```

Comment:
As discussed, it seems in-memory storage is shared between all sessions, so the value above would be unique for all current users and continually change, which would result in excessive calls to refresh user details. Suggestion would be to use a different session key which does not get updated with user details:

```suggestion
        req.session.userVersion = res.headers['x-cas-user-version']
```

Reply:
As discussed, the above isn't doable directly -- this is a call to the API using Superagent, so the `req` object here is not the same as the user session!